### DISCUSSION TITLE
Gradle Build Failed

### DISCUSSION THREAD

**Discussion by sivasubramaniamv** [oppia/oppia-android#4966] [Q&A (Installation)] [answered]
### Operating System

Windows 11

### Android Studio Version

Flamingo Stable Release

### RAM size

8 GB

### Free disk space

44 GB

### Which step are you stuck on?

I cloned the repo and followed the instructions as mentioned in the [installation guide](https://github.com/oppia/oppia-android/wiki/Installing-Oppia-Android). But the build failed.

By default, the project was using a JDK called azul-15 (C:\Users\sivam\.jdks\azul-15.0.10) which got installed at the same time the project was cloned. The build failed with multiple errors.

The guide instructs using Java 8. I downloaded [**Java 8u202**](https://www.oracle.com/in/java/technologies/javase/javase8-archive-downloads.html) from the Oracle website and configured it in the IDE:
**Settings -> Build, Execution, Deployment -> Build Tools -> Gradle -> Gradle JDK**.

I set the same JDK in the PATH and JAVA_HOME environment variables as well.

When I rebuilt the project multiple errors of the same type (inferable from the error log) remained. I found a possible [**solution**](https://stackoverflow.com/a/24568901/17240756) in StackOverflow, but it requires modifying the module-level gradle file.

Can someone developing on Windows help me out?

**Note:** Android Studio also warned that the project is using a very old AGP version (3.6.1) which will not be supported in future Android Studio releases. It suggested upgrading AGP to at least version 4.0.0.

### Error Log

```bash
C:\Users\sivam\AndroidStudioProjects\oppia-android\model\build\generated\source\proto\main\java\org\oppia\android\app\model\OppiaLanguage.java:25: error: unmappable character for encoding Cp1252
   * Corresponds to the Arabic (اَلْعَرَب�?يَّة�?‎) macro language. IETF BCP 47 language tag: ar.
                                                   ^
C:\Users\sivam\AndroidStudioProjects\oppia-android\model\build\generated\source\proto\main\java\org\oppia\android\app\model\OppiaLanguage.java:25: error: unmappable character for encoding Cp1252
   * Corresponds to the Arabic (اَلْعَرَب�?يَّة�?‎) macro language. IETF BCP 47 language tag: ar.
                                                             ^
C:\Users\sivam\AndroidStudioProjects\oppia-android\model\build\generated\source\proto\main\java\org\oppia\android\app\model\OppiaLanguage.java:41: error: unmappable character for encoding Cp1252
   * Corresponds to the Hindi (हिन�?दी) macro language. IETF BCP 47 language tag: hi.
                                          ^
C:\Users\sivam\AndroidStudioProjects\oppia-android\model\build\generated\source\proto\main\java\org\oppia\android\app\model\OppiaLanguage.java:92: error: unmappable character for encoding Cp1252
   * Corresponds to the Arabic (اَلْعَرَب�?يَّة�?‎) macro language. IETF BCP 47 language tag: ar.
                                                   ^
C:\Users\sivam\AndroidStudioProjects\oppia-android\model\build\generated\source\proto\main\java\org\oppia\android\app\model\OppiaLanguage.java:92: error: unmappable character for encoding Cp1252
   * Corresponds to the Arabic (اَلْعَرَب�?يَّة�?‎) macro language. IETF BCP 47 language tag: ar.
                                                             ^
C:\Users\sivam\AndroidStudioProjects\oppia-android\model\build\generated\source\proto\main\java\org\oppia\android\app\model\OppiaLanguage.java:108: error: unmappable character for encoding Cp1252
   * Corresponds to the Hindi (हिन�?दी) macro language. IETF BCP 47 language tag: hi.
                                          ^
```

**Comment by MohitGupta121** [comment]
Hello @siva-subramaniam-v, Can you try to Invalidate cache and restart in Android studio. Or re-clone the project in your local. I think you missing something initially.
We suggest you to download bumblebee version android version from [here](https://developer.android.com/studio/archive)
If you get any error feel free to ask here.

**Reply by sivasubramaniamv** [reply]
Isn't bumblebee too old to be suggested?

Ideally the project should be adapted to changes in new Android Studio versions rather than sticking to a very old version.

**Reply by MohitGupta121** [reply]
Actually due to gradle version and kotlin version of oppia-android most things work in that version, but after gradle to bazel of oppia-android this problem got solved.

But I suggest you to first try re-clone and follow all instructions again.
Thanks!

**Comment by adhiamboperes** [comment, marked as answer]
In Android Studio, the default file encoding can be determined by the operating system's default encoding. Android Studio follows the default encoding set by the underlying platform to handle file encoding for various operations, such as reading and writing files.

On different operating systems, the default file encoding may vary:

Windows: The default file encoding on Windows is typically Cp1252 (also known as Windows-1252).

macOS: The default file encoding on macOS is usually UTF-8.

Linux: On Linux distributions, the default file encoding can vary depending on the distribution and system configuration. However, UTF-8 is widely used and often set as the default encoding.

You can change your file encoding to UTF-8 by visiting `Android Studio Settings > Editor > File Encodings`

<img width="1080" alt="Screenshot 2023-05-15 at 13 31 09" src="https://github.com/oppia/oppia-android/assets/59600948/878eee67-07b6-4fb9-8101-c1c98e931917">

**Comment by adhiamboperes** [comment]
Please refer to https://github.com/oppia/oppia-android/discussions/4984#discussioncomment-5931618

**Comment by adhiamboperes** [comment]
Important to note that while this message will commonly show up while building on windows, the project should still build correctly.
